tbtadm/controller.cpp

```diff
@@ -653,6 +655,188 @@ catch (...)
     m_err << "Unknown exception\n";
 }
 
+void tbtadm::Controller::addToBootACL(const fs::path& dir, std::string uuid)
+{
+    auto fname = dir.filename().string();
+    auto num = fname.substr(0, fname.find('-'));
+    auto boot_acl_path = sysfsDevicesPath / (domain + num) / boot_acl;
+
+    if (!fs::exists(boot_acl_path))
+    {
+        /* Cannot find boot_acl */
+        return;
+    }
+
+    /* Read boot_acl */
+    std::stringstream stream(readAndTrim(boot_acl_path));
+    /* Check maximum allowed boot_acl string size */
+    if (stream.str().size() > 36 * 16 + 1 * 15)
```

Comment:
1. I'm not sure what we are trying to protect against here.
1. Do we really want to hard-code 16 here? Why not use the `boot_acl` content to learn how many entries the HW supports?

Reply:
I check here boot_acl (since it can be constructed like in my test case with simulated thunderbolt).
@westeri can boot_acl be of different size then 16 entries?

Reply:
In theory, yes. The reason why boot_acl always shows exact amount of entries (even if empty) is just because then userspace can learn how many entries the given controller supports.

Reply:
But I still need some number to check boundaries.

Reply:
You parse the string you get from `boot_acl` from sysfs, and the count of commas + 1 is the amount of possible entries. Not sure what else you need to check here (especially as this is `std::string`, not something that you have to manually allocate its memory).

Reply:
But I still need some check for the input, right?